lib/ui/form/widget/form_survey_answer_rating_emoji.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_survey/theme/app_colors.dart';
+import 'package:flutter_survey/theme/app_dimensions.dart';
+
+final _selectedEmojiRatingIndexProvider = StateProvider.autoDispose<int>(
```

Comment:
It seems like the previous selection is transferred to the next widget because we're using the same provider for all the widgets. What if we instead move it inside the class, then it will create a new provider for every widget. If you agree, could you please do the same for `FormSurveyAnswerSmiley`? 🙏   

https://user-images.githubusercontent.com/18277915/234237523-cf9dec0f-fcdc-449f-807c-c93954f60421.mov